resources/js/features/root/components/navbar/NavigationSection.tsx

```diff
@@ -0,0 +1,25 @@
+import {AccessBarrier, AccessBarrierProps} from "components/utils";
+import {Component, For, Show, splitProps} from "solid-js";
+import {NavigationItem, NavigationItemProps} from "./NavigationItem";
+
+export interface NavigationSectionProps
+  extends Omit<AccessBarrierProps, "children" | "queries" | "Fallback" | "Error" | "Pending"> {
```

Comment:
To jest naprawdę paskudne :) im więcej tam rzeczy wymienione tym gorsze. Przecież jak to czytam to chyba chcę wiedzieć co trzeba przekazać do tych propsów a nie czego nie trzeba. A jeszcze ten typ z którego usuwasz też jest zdefiniowany przez Omit, masakra.

Zresztą takie coś jak Fallback, Error i Pending to można po prostu mergeProps podać defaulty zamiast zabraniać. Ale można też zabronić, tylko nie w taki sposób. Już prędzej Pick ale lepiej explicite.

Dowodem na to jak się trudno to czyta jest to, że sam nie zauważyłeś, że ten typ nie ma children. Pick by ci chociaż sprawdził czy tam nie podajesz kluczy bez sensu.

Reply:
Jak żem pisał gdzieś indziej, propsy mi się nie wydają takie przezroczyste (nie licząc może przekazywania propsów do natywnych elementów), tylko wolałbym wymienić co jest potrzebne. A jak już skorzystać z typu to Pick. Bo wtedy mam "z tego typu proszę chleb, masło i dżem", a z Omit to masz "poproszę ten typ ale bez młyna, krowy, drzewa i dzieci".